src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -111,6 +118,43 @@ public void setPerson(Person target, Person editedPerson) {
         addressBook.setPerson(target, editedPerson);
     }
 
+    //=========== Selected Person ===========================================================================
+
+    @Override
+    public ReadOnlyProperty<Person> selectedPersonProperty() {
+        return selectedPerson;
+    }
+
+    @Override
+    public Person getSelectedPerson() {
+        return selectedPerson.getValue();
+    }
+
+    @Override
+    public void setSelectedPerson(Person person) {
+        if (person != null && !filteredPersons.contains(person)) {
+            throw new PersonNotFoundException();
+        }
```

Comment:
```suggestion
        requireNonNull(person);
        if (!filteredPersons.contains(person)) {
            throw new PersonNotFoundException();
        }
```
Could consider doing this to match what the module did, but I have no issues with the current implementation either.

Reply:
Thanks for pointing out!